File: backend/src/main/java/com/avaliaplus/service/TurmaService.java
package com.avaliaplus.service;

import com.avaliaplus.model.Turma;
import com.avaliaplus.repository.TurmaRepository;
import jakarta.persistence.EntityNotFoundException;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import java.util.List;

@Service
@RequiredArgsConstructor
public class TurmaService {

    private final TurmaRepository turmaRepository;

    @Transactional
    public Turma criar(Turma turma) {
        return turmaRepository.save(turma);
    }

    public List<Turma> listar() {
        return turmaRepository.findAll();
    }

    public Turma buscarPorId(Long id) {
        return turmaRepository.findById(id)
                .orElseThrow(() -> new EntityNotFoundException("Turma não encontrada"));
    }

    @Transactional
    public Turma atualizar(Long id, Turma dados) {
        Turma turma = buscarPorId(id);
        turma.setNome(dados.getNome());
        turma.setAno(dados.getAno());
        turma.setCurso(dados.getCurso());
        return turmaRepository.save(turma);
    }

    @Transactional
    public void deletar(Long id) {
        turmaRepository.deleteById(id);
    }
}

File: frontend/src/services/turmaService.ts
import api from './api';
import type { Turma } from '../types';

export const turmaService = {
  listar: () => api.get<Turma[]>('/turmas').then((r) => r.data),
  buscar: (id: number) => api.get<Turma>(`/turmas/${id}`).then((r) => r.data),
  criar: (data: unknown) => api.post<Turma>('/turmas', data).then((r) => r.data),
  atualizar: (id: number, data: unknown) => api.put<Turma>(`/turmas/${id}`, data).then((r) => r.data),
  deletar: (id: number) => api.delete(`/turmas/${id}`),
};
